src/app/views/sidebar/resource-explorer/ResourceExplorer.tsx

```diff
@@ -20,23 +21,21 @@ import { translateMessage } from '../../../utils/translate-messages';
 import { classNames } from '../../classnames';
 import { NoResultsFound } from '../sidebar-utils/SearchResult';
 import { sidebarStyles } from '../Sidebar.styles';
-import { UploadPostmanCollection } from './collection/UploadCollection';
-import CommandOptions from './command-options/CommandOptions';
-import {
-  createResourcesList, getResourcePaths,
-  getUrlFromLink
-} from './resource-explorer.utils';
+import { createResourcesList, getResourcePaths, getUrlFromLink } from './resource-explorer.utils';
 import ResourceLink from './ResourceLink';
-import { navStyles } from './resources.styles';
+import { navStyles, resourceExplorerStyles } from './resources.styles';
+import { usePopups } from '../../../services/hooks/usePopups';
 
 const UnstyledResourceExplorer = (props: any) => {
-  const { resources: { data, pending }, collections } = useAppSelector(
+  const { resources: { data, pending }, collections: {collections} } = useAppSelector(
     (state) => state
   );
```

Comment:
```suggestion
  const resources = useAppSelector(state=>state.resources)
  const collections = useAppSelector(state=>state.collections)
```

Get the individual state values from the state explicitly. When you get the entire state object and de-structure, you keep getting re-renders of components that don't need to re-render just because the current state has changed.